### PR TITLE
[AOTI] add float support of triton

### PR DESCRIPTION
**Pull request by sijiac**
Summary: As the title

Test Plan: buck2 test 'fbcode//mode/opt' fbcode//caffe2/test/dynamo:test_dynamo -- --exact 'caffe2/test/dynamo:test_dynamo - test_functions.py::DefaultsTests::test_triton_kernel_None_arg' --print-passing-details

Differential Revision: D51421325

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: test/dynamo/test_functions.py [hunk 1]
```diff
@@ -1995,11 +1995,12 @@ def call_triton(x: torch.Tensor):
 
     @requires_cuda()
     @skipIfRocm
-    def test_triton_kernel_None_arg(self):
+    def test_triton_kernel_None_Float_arg(self):
```
Comment (oulgen) [inline test/dynamo/test_functions.py:1998]: test_triton_kernel_various_args

**Review by oulgen — APPROVED** [review]

**Comment by oulgen** [comment]
I landed my changes, please rebase past https://github.com/pytorch/pytorch/pull/114002

**Comment by sijiac** [comment]
rebase to the master branch

**Review by aakhundov — APPROVED** [review]

**Pull request closed without merging**